src/transformers/file_utils.py

```diff
@@ -1907,31 +1907,61 @@ def copy_func(f):
     return g
 
 
+def is_local_clone(repo_path, repo_url):
```

Comment:
This could move to the `huggingface_hub` library.

Reply:
Yes, we could refactor that in a utility as it's already getting used in the proposed solution in https://github.com/huggingface/huggingface_hub/pull/132

Reply:
Btw here I'm checking the remotes to see if a repository is linked to another, but we could theoretically imagine two repositories with dissimilar remotes that would still be linked by history.

A better way to check this would be to check the commit history directly and check if they align - but I found the remote check simpler, and I believe it's enough of a check for our use-case.